### ISSUE TITLE
User can schedule a migration plan

### ISSUE THREAD

**Issue by serenamarie125** [ManageIQ/miq_v2v_ui_plugin#369]
The user should be able to schedule a migration plan, by specifying a date and time for it to be run.

**Comment by serenamarie125** [comment]
Note that there is a requirements meeting around schedule the first week of June.  Design work won't be started until we know requirements and identified use cases.

**Comment by serenamarie125** [comment]
@bthurber, I'm guessing the end goal is to be able to both:
1. Schedule an existing migration plan
1. Create a migration plan and specify the date and time for it to be run

For sprint 9, is (1) sufficient?

**Comment by bthurber** [comment]
@serenamarie125 yes that sounds like it will meet the criteria.  The only other option I would suggest is to be able to delete a scheduled migration before it has run.

**Comment by serenamarie125** [comment]
@vconzola please add mockups for scheduling a migration plan

**Comment by serenamarie125** [comment]
Good news @bdunne thanks for the info!  Removing needs-api message.

**Comment by vconzola** [comment]
@serenamarie125 I checked in ClodForms and they do scheduling like this:
![adding a new schedule in cfme](https://user-images.githubusercontent.com/22823035/41055188-8b74b2ac-698e-11e8-88bb-c71b02df27e7.png)
I also talked with Matt C. and the PR for the PatterFly core date/time picker that Agnes was working on will be merged very soon.  He said, "Yes, the plan is to put this into PF Core. It leverages the current Bootstrap widget, so there is minimal dev work to be done. You should feel free to use this approach and let @nding-anges and I know if you see any issues with applying this." So which way should we go - stay consistent with current CloudForms or implement PatternFly component?

**Comment by vconzola** [comment]
@serenamarie125 I have some mockups done, but want to review them with you and team before adding here. Can we discuss in Thursday standup?

**Comment by nding-anges** [comment]
@vconzola Hi, we will send a P.R of Date&Time Picker soon. Please find more detail about the design here: https://docs.google.com/document/d/1kVAhNzSB-szucfeGgq815dB6e214WavZ-gqRR0Sydj0/edit?usp=sharing  and let us know if you have any questions.

**Comment by vconzola** [comment]
Mockups are here: https://docs.google.com/presentation/d/1gMpkoVBzYb_L5orh_VFk3ZhwtLfl1S6fuI2whIdJVh8/edit?usp=sharing
Still a WIP, but enough to get started on development.

**Comment by AllenBW** [comment]
just cuz i like 🖼 

<img width="864" alt="screen shot 2018-06-08 at 12 36 55 pm" src="https://user-images.githubusercontent.com/6640236/41170063-e2b3b5fa-6b18-11e8-9717-402aff4fcf30.png">
<img width="865" alt="screen shot 2018-06-08 at 12 36 47 pm" src="https://user-images.githubusercontent.com/6640236/41170064-e2bfba58-6b18-11e8-9178-d0e0490c8b05.png">
<img width="867" alt="screen shot 2018-06-08 at 12 36 37 pm" src="https://user-images.githubusercontent.com/6640236/41170066-e2ca98a6-6b18-11e8-94c3-49fd9c17eb54.png">

**Comment by serenamarie125** [comment]
@vconzola I think we should pull Stacy in for messaging around the cancel.  It feels like this is actually canceling the plan which is migrating, when it's actually "unscheduling" it.  Maybe "remove schedule" or "unschedule" as the action button label would be more descriptive?

**Comment by mturley** [comment]
This all looks good so far, but I just want to throw this out there before it gets discussed from a design standpoint: If we are asked to build a calendar of upcoming migrations, I fear it will be more work than we expect. I've just been spooked by https://zachholman.com/talk/utc-is-enough-for-everyone-right 😄 

I agree with @serenamarie125 about the word "cancel" maybe sounding too destructive, too.

**Comment by AllenBW** [comment]
@vconzola Gonna echo the above sentiments, what is the intended behavior of the cancel? To clear a migration planned in the future?

**Comment by vconzola** [comment]
@serenamarie125 @mturley @AllenBW All right! All right! I get it. "Cancel" was a crappy word choice. Stop ganging up on me.  :-)  I really don't like "unschedule" either and I think "remove" has the same problem as "cancel". But I can't think of a better alternative, so let's try "unschedule".  It's possible this will get revisited when Stacy does her text review.  Here are updated mockups.
![filter is not started - w schedule scheduled w cancel _2](https://user-images.githubusercontent.com/22823035/41248773-fea0822a-6d7f-11e8-80e0-db421d057bee.png)
![filter is not started - w schedule scheduled w cancel confirm _2](https://user-images.githubusercontent.com/22823035/41248788-078ae27c-6d80-11e8-85c3-314623af6606.png)

**Comment by vconzola** [comment]
@serenamarie125 Any guidance on which scheduling components we should use for the actual schedule creation - the one about to be merged into PatternFly core or the ones currently used in CloudForms? See my comment above from last week.

**Comment by AllenBW** [comment]
this body results in success 

```
{
  "action" : "edit",
  "resource" : {
    "options" : {
      "schedule_type":"schedule", 
	  "schedule_time": 1528821391
    }
  }
}
```

**Comment by AllenBW** [comment]
soooooooooooo

when posting with a body of 👇 
```
{
  "action" : "edit",
  "resource" : {
    "options" : {
      "schedule_type":"schedule", 
      "schedule_time": 1528821391
    }
  }
}
```

We overwrite the contents of the service template options... looks like put and patch, while are listed as actions aren't enabled? @bdunne do yah see anything that might confirm or deny the (un)availability of these additional actions?

**Comment by AllenBW** [comment]
Yah know... we could really use... https://github.com/patternfly/patternfly-react/issues/232  🤔 💡

**Comment by vconzola** [comment]
Hi @nding-anges - Has the P.R of Date&Time Picker been merged yet?

**Comment by nding-anges** [comment]
Hi @vconzola We have just fixed one little issue and the P.R is waiting for review now. https://github.com/patternfly/patternfly-design/pull/676

**Comment by priley86** [comment]
hey @vconzola @AllenBW - some things we have to consider here. Since we don't yet have this component in PF React and there is a new design coming...here is my advice:

* there is existing jquery datepickers and timepickers in the existing design and the new design (and existing implementations for datepicker and timepicker independently). The new design for integrated datepicker/timepicker will still have to be developed. For that reason, I would propose fast tracking this by showing the existing design if we want to be 100% PatternFly.

PF Jquery Datepicker:
https://rawgit.com/patternfly/patternfly/master-dist/dist/tests/bootstrap-datepicker.html

PF Jquery Timepicker:
https://rawgit.com/patternfly/patternfly/master-dist/dist/tests/time-picker.html

Some alternatives in React that are similar, but not 100%:

React Bootstrap Datepicker (not in line w/ existing design 100% but similar):
https://github.com/pushtell/react-bootstrap-date-picker

React Widget Date & Time Picker (not in line w/ new design but would work):
http://jquense.github.io/react-widgets/api/DateTimePicker/

There may be others that could fit, but we really need this component still in PF React. I would pick a short term option that works for now and we can start the contribution to PF React separately that meets the new design and circle back later when it's done. 

thoughts?

**Comment by AllenBW** [comment]
down with React Widget Date & Time Picker!  would be real hesitant to introduce a jquery asset... but just need the blessing that we can go our own way (for the time being) 🙇 🙏

**Comment by vconzola** [comment]
@AllenBW After discussing with @priley86 I say we go with the existing jquery assets to stay PatternFly compliant and in parallel try to get a React component developed based on the design  [here:](https://github.com/patternfly/patternfly-design/issues/459#issuecomment-394842964), the PR for which should be merged very soon. Patrick said it will take a while to implement the design in React to stay PF compliant. I know you don't agree, but do you have what you need to do the jquery date/time picker or do you need a mockup?

**Comment by AllenBW** [comment]
Yeah! mockups for the record would be gooood to have @vconzola  🙇‍♀️

**Comment by AllenBW** [comment]
@vconzola 
<img width="681" alt="screen shot 2018-06-18 at 11 17 20 am" src="https://user-images.githubusercontent.com/6640236/41545810-08943c58-72ea-11e8-8f3f-474b17fa1a13.png">
<img width="687" alt="screen shot 2018-06-18 at 11 17 12 am" src="https://user-images.githubusercontent.com/6640236/41545811-08bc5b52-72ea-11e8-83c2-26d2afdc7fa8.png">

no mockups required if this is what yah had in mind 🤔

**Comment by serenamarie125** [comment]
@AllenBW @vconzola  i'm checking with Jenny on the spacing ... i'll come back with some direction there in a couple!

**Comment by serenamarie125** [comment]
@AllenBW left margin should be 20px so that it lines up with the modal heading.  There should be 15 px between the date and time picker components.  @vconzola fyi i did verify that with Jenny from a PF perspective

**Comment by vconzola** [comment]
@AllenBW Sorry I've been slow to respond. Trying to catch up from meetings in Westford last week. The only thing I see missing/different from the current CloudForms scheduler is a time zone selector. (See screen shot way at the top of this issue.)

**Comment by vconzola** [comment]
@AllenBW One other thing...title should be Schedule Migration Plan.

**Comment by AllenBW** [comment]
@vconzola can we get updated mocks?

**Comment by vconzola** [comment]
@AllenBW I've given this some thought and looked at various date/time pickers in jQuery and I like what you did above as much as any of them. So let's just use what you did but with the correct title "Schedule Migration Plan" and with the spacing specified by Serena above. We will not add a time zone selector since I think it will just add confusion. We display times in other places in the UI, which I assume are based on the platform clock. So the scheduler should just use the same clock reference.
<img width="881" alt="screen shot 2018-06-18 at 4 36 34 pm" src="https://user-images.githubusercontent.com/22823035/41627635-12dafc78-73ef-11e8-9be8-33fc835ca923.png">

**Comment by JPrause** [comment]
@miq-bot add_label blocker
This is a blocker to: https://bugzilla.redhat.com/show_bug.cgi?id=1564255

**Comment by JPrause** [comment]
Any update on this PR. We have an upcoming build on July 10

**Comment by AllenBW** [comment]
#375 is the pr for this issue, has the latest info, presently blocked by api work (detailed in pr) @JPrause

**Comment by nding-anges** [comment]
@vconzola Hi， the P.R of Date&Time Picker been merged. https://rawgit.com/patternfly/patternfly/master-dist/dist/tests/date-and-time.html#!
